RFC/RFC-14-p2p-adapter.md

```diff
@@ -0,0 +1,348 @@
+---
+layout: doc
+rfc: 14
+date: 2022-10-17
+title: P2P adapter with routing service
+status: DRAFT
+authors:
+  - hugoArregui
+  - agusAldasoro
+---
+
+## Abstract
+
+This document describes a possible implementation for a P2P adapter using a fallback messaging service. For more information about what and adapter is, check [ADR-81](/ADR/ADR-81-minimum-comms-transport.md).
+
+## Need
+
+A P2P adapter is needed in order to provide a light mechanism for comms (compared to [RFC-5: comms service using websocket](/RFC/RFC-5-ws-comms.md)). This adapter should guarantee the delivery of a message to any peer in the network (or cluster, [real-time clustering of users in Islands (ADR-35)](/adr/ADR-35)).
+
+## Approach
+
+This approach is simple, it should be easy to debug and it could be refined in the future for performance if needed. 
+
+This implementation requires two extra services:
+
+- Messaging service: in charge of sending messages between specific peers.
+- Routing service: in charge of keeping track of the connections between peer and given them back routing information.
+
+As defined in [ADR-81](/ADR/ADR-81-minimum-comms-transport.md), each peer will known the ids of the peers around them using the information provided by the CommunicationsDirector.
+
+The basic idea of this implementation is for peers to connect randomly to each other forming a mesh, and reporting their connections to the routing service. The routing service will build routing tables for each peer to connect to every other. Each peer will receive from the routing service the path to all other conected peers. When a peer delivers a message, it will use the path provided by the routing table, and if there is no one, it will fallback to the messaging service.
```

Comment:
```suggestion
The basic idea of this implementation is for peers to connect randomly to a subset of the others forming a mesh, and reporting their connections to the routing service. The routing service will build routing tables for each peer to connect to every other. Each peer will receive from the routing service the path to all other connected peers. When a peer needs to deliver a message, it will use the path provided by the routing table, and if there is not any, it will fallback to the messaging service.
```